code/4ed_render_target.cpp

```diff
@@ -152,7 +152,22 @@ draw_rectangle_outline(Render_Target *target, Rect_f32 rect, f32 roundness, f32
         vertices[i].color = color;
         vertices[i].half_thickness = half_thickness;
     }
-    draw__write_vertices_in_current_group(target, vertices, ArrayCount(vertices));
+    
+    // NOTE simon (03/04/24): If any vertex is in the render target bounds, we draw the rectangle.
+    // It would be better to use the current clip rect, but it's not available here. We could pass
+    // it down in the function signature if necessary.
+    // This is not in the loop to try minimize the impact it could have on performance when there
+    // are a lot of vertex (we don't test duplicated vertex positions).
+    Rect_f32 target_rect = Rf32( 0, 0, ( f32 ) target->width, ( f32 ) target->height );
+    b32 draw = false;
+    draw = draw || rect_contains_point( target_rect, vertices[ 0 ].xy );
+    draw = draw || rect_contains_point( target_rect, vertices[ 1 ].xy );
+    draw = draw || rect_contains_point( target_rect, vertices[ 2 ].xy );
+    draw = draw || rect_contains_point( target_rect, vertices[ 5 ].xy );
```

Comment:
These should not be 5 right?

Reply:
Although maybe its better to use `rect` rather than vertices so its a little more obvious. We could also do it first and early out? avoiding basically all the work in here?

Reply:
Also should point out there is already a function to do this:
```c
function b32
rect_overlap(Rect_i32 a, Rect_i32 b){
    return(range_overlap(rect_range_x(a), rect_range_x(b)) &&
           range_overlap(rect_range_y(a), rect_range_y(b)));
}
```

Reply:
More about that in the following reply.